confluent_platform/datadog_checks/confluent_platform/data/metrics.yaml

```diff
@@ -0,0 +1,1276 @@
+# Default metrics collected by this check. You should not have to modify this.
+jmx_metrics:
+  # Kafka Broker Metrics (Yammer Metric Gauge)
+  - include:
+      bean:
+        # Size of the request queue. A congested request queue will not be able to process incoming or outgoing requests
+        - kafka.network:type=RequestChannel,name=RequestQueueSize
+        # Average fraction of time the network processor threads are idle. Values are between 0 (all resources are used) and 1 (all resources are available)
+        - kafka.network:type=SocketServer,name=NetworkProcessorAvgIdlePercent
+        # Number of requests waiting in the fetch purgatory. This is high if consumers use a large value for fetch.wait.max.ms .
+        - kafka.server:type=DelayedOperationPurgatory,name=PurgatorySize,delayedOperation=Fetch
+        # Number of requests waiting in the producer purgatory. This should be non-zero when acks=all is used on the producer.
+        - kafka.server:type=DelayedOperationPurgatory,name=PurgatorySize,delayedOperation=Produce
+        # Maximum lag in messages between the follower and leader replicas. This is controlled by the `replica.lag.max.messages` config.
+        - kafka.server:type=ReplicaFetcherManager,name=MaxLag,clientId=Replica
+        # Number of leaders on this broker. This should be mostly even across all brokers. If not, set auto.leader.rebalance.enable to true on all brokers in the cluster.
+        - kafka.server:type=ReplicaManager,name=LeaderCount
+        # Number of partitions on this broker. This should be mostly even across all brokers.
+        - kafka.server:type=ReplicaManager,name=PartitionCount
+        # Number of partitions whose in-sync replicas count is less than minIsr.
+        - kafka.server:type=ReplicaManager,name=UnderMinIsrPartitionCount
+        # Number of under-replicated partitions (| ISR | < | all replicas |). Alert if value is greater than 0.
+        - kafka.server:type=ReplicaManager,name=UnderReplicatedPartitions
+      attribute:
+        Value:
+          alias: $domain.$type.$name
```

Comment:
What does a typical `$domain` look like? Eg is it `io.confluent.[...]`? I'm asking because the `metric_prefix` is currently set to `confluent.`, so wondering if the prefix will be respected in all cases here.

Reply:
The domain correction to the first part of the bean name, in this case: `kafka.server`.

The initial intent was to use the domain as bean prefix to keep compatibility with Kafka Integration. But since some metrics names are now different, that's not a valid reason anymore. So we can prefix all metrics with `confluent.`.

Reply:
Related commit https://github.com/DataDog/integrations-core/pull/5733/commits/0d698ddd529e07425d4325a3ea0c20d6d6cc2403